plum/resolver.py

```diff
@@ -66,62 +23,62 @@ class Resolver:
         is_faithful (bool): Whether all signatures are faithful or not.
     """
 
-    def __init__(self):
-        self.signatures: List[Signature] = []
-        self.is_faithful: bool = True
+    def __init__(self, signatures: Iterable[Signature]):
+        signatures_dict = {hash(s): s for s in signatures}
```

Comment:
Ahhh, I now understand that this is used to remove duplicate signatures. What I'm worried about is specifically which duplicate is kept. Specifically, if `signatures` implements the same method twice, then the _later_ one should be kept. Right now, which implementation is kept is up to the implementation of the dictionary. I think it's safest to implement this in way that very explicitly retains the later implementations.

Reply:
We should also have this tested by a unit test!